src/Microsoft.Restier.Providers.EntityFramework/Model/ModelProducer.cs

```diff
@@ -62,7 +62,19 @@ public Task<IEdmModel> GetModelAsync(ModelContext context, CancellationToken can
             var dbContext = context.GetApiService<DbContext>();
 
             var efModel = (dbContext as IObjectContextAdapter).ObjectContext.MetadataWorkspace;
-            var efEntityContainer = efModel.GetItems<EntityContainer>(DataSpace.CSpace).Single();
+            var efEntityContainers = efModel.GetItems<EntityContainer>(DataSpace.CSpace);
+            var efEntityContainer = efEntityContainers.FirstOrDefault(c => c.Name == dbContext.GetType().Name);
+            if (efEntityContainer == null)
+            {
+                if (efEntityContainers.Count > 1)
+                {
```

Comment:
Yes, I know which case there will be more than one container, that's why I had to write this code. If you register 2 APIs with 2 different DB contexts in the same website, then the call to `efModel.GetItems<EntityContainer>(DataSpace.CSpace)` will return more than one container, even though they are in separate EDMX files in separate DbContexts. One would think that wouldn't be the case, so maybe this exposed a flaw somewhere deeper in the OData or EF framework. But that is OK, because the code above these lines *should* always return the right container for the particular context. This particular set of highlighted code is only there in case, for some presently unknown reason, the query to get the matching context fails, It's a defensive measure to help developers understand what is going on, instead of the current code just assuming everything will work, until it doesn't, and not giving you any clue as to why.